### PR TITLE
Task/store init from window

### PR DESCRIPTION
**Pull request by paulmskim**
[TEC-3339]

This task is the first task of setting up allowing us to set the initial state of our block editor application through the reducer rather than via mounting of the blocks

Related to:
* https://github.com/moderntribe/tribe-common/pull/1317
* https://github.com/moderntribe/events-pro/pull/1522

[TEC-3339]: https://moderntribe.atlassian.net/browse/TEC-3339

### CODE REVIEW

File: src/Tribe/Editor/Configuration.php [hunk 1]
```diff
@@ -15,6 +16,45 @@ public function hook() {
...
 	}
 
+	/**
+	 * Setup the initial Event Object.
+	 *
+	 * @since TBD
+	 *
+	 * @todo @backend Improve this method and verify if this is the correct place for this to live.
+	 *
+	 * @param null|int|WP_Post $event
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: :no_entry_sign: Missing parameter comment

<sub>hash: 26b0dcea55a08f266dcc1b4b070cbc8d</sub>

File: src/Tribe/Editor/Configuration.php [hunk 2]
```diff
@@ -15,6 +16,45 @@ public function hook() {
...
+
+		if ( ! $event ) {
+			$event = tribe_get_request_var( 'post', false );
+		}
+
+		if ( $event ) {
+			$event = get_post( $event );
+		}
+
+		if ( ! $event instanceof  WP_Post) {
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: * :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: Expected 1 space after "instanceof"; 2 found
* :no_entry_sign: No space before closing parenthesis is prohibited
* :no_entry_sign: No space before closing parenthesis is prohibited

<sub>hash: 5a349b4f160ef1c36a39ebd239e36bfe</sub>

File: src/Tribe/Editor/Configuration.php [hunk 3]
```diff
@@ -15,6 +16,45 @@ public function hook() {
...
+			return $data;
+		}
+
+		$meta = get_post_meta( $event->ID );
+		$meta = array_map( static function( $value ) {
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: b189c8b99d0bf5ca0912e0856e1320d1</sub>

File: src/Tribe/Editor/Configuration.php [hunk 4]
```diff
@@ -15,6 +16,45 @@ public function hook() {
...
+			return count( $value ) === 1 ? reset( $value ) : $value;
+		}, $meta );
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: * :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 3e1efe2ff34e5f4d228ca36bb1290cde</sub>

File: src/Tribe/Editor/Configuration.php [hunk 5]
```diff
@@ -15,6 +16,45 @@ public function hook() {
...
+
+		$data['is_new_post'] = false;
+		$data['meta'] = $meta;
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: :warning: Equals sign not aligned with surrounding assignments; expected 8 spaces but found 1 space

<sub>hash: ca0e810cdd0274eb6520bcccd3614f00</sub>

File: src/Tribe/Editor/Configuration.php [hunk 6]
```diff
@@ -24,8 +64,12 @@ public function hook() {
 	 * @return array
 	 */
 	public function editor_config( $editor_config ) {
-		$tec = empty( $editor_config['events'] ) ? array() : $editor_config['events'];
+		$tec = empty( $editor_config['events'] ) ? [] : $editor_config['events'];
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: :warning: Equals sign not aligned with surrounding assignments; expected 21 spaces but found 1 space

<sub>hash: 05f1ae9c2ad8e9496d259bd5b1efacab</sub>

File: src/Tribe/Editor/Configuration.php [hunk 7]
```diff
@@ -24,8 +64,12 @@ public function hook() {
...
 		$editor_config['events'] = array_merge( (array) $tec, $this->localize() );
+
+		$post_objects = empty( $editor_config['post_objects'] ) ? [] : $editor_config['post_objects'];
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: :warning: Equals sign not aligned with surrounding assignments; expected 18 spaces but found 1 space

<sub>hash: 9181743cb1a86bea982f5cb70d916fb5</sub>

File: src/Tribe/Editor/Configuration.php [hunk 8]
```diff
@@ -39,15 +83,14 @@ public function editor_config( $editor_config ) {
 	public function localize() {
 		/** @var Tribe__Events__Admin__Event_Meta_Box $events_meta_box */
 		$events_meta_box = tribe( 'tec.admin.event-meta-box' );
-		return array(
+		$data = [
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: :warning: Equals sign not aligned with surrounding assignments; expected 12 spaces but found 1 space

<sub>hash: 56cbbb89b480fe857629dc0cce1fdcae</sub>

File: src/Tribe/Editor/Configuration.php [hunk 9]
```diff
@@ -39,15 +83,14 @@ public function editor_config( $editor_config ) {
...
 			'settings'      => tribe( 'events.editor.settings' )->get_options(),
 			'timezoneHTML'  => tribe_events_timezone_choice( Tribe__Events__Timezones::get_event_timezone_string() ),
-			'priceSettings' => array(
+			'priceSettings' => [
 				'defaultCurrencySymbol'   => tribe_get_option( 'defaultCurrencySymbol', '$' ),
 				'defaultCurrencyPosition' => (
 					tribe_get_option( 'reverseCurrencyPosition', false ) ? 'suffix' : 'prefix'
-				),
-			),
+				), ],
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php]: * :no_entry_sign: Closing parenthesis of array declaration must be on a new line
* :no_entry_sign: Array closer not aligned correctly; expected 12 space(s) but found 19
* :no_entry_sign: Closing parenthesis of array declaration must be on a new line

<sub>hash: 367e36be5f63104edb8b30285f05cc9f</sub>

File: src/Tribe/Editor/Configuration.php [hunk 10]
```diff
@@ -24,8 +28,17 @@ public function hook() {
...
 	public function editor_config( $editor_config ) {
-		$tec = empty( $editor_config['events'] ) ? array() : $editor_config['events'];
+		$tec                     = empty( $editor_config['events'] ) ? [] : $editor_config['events'];
 		$editor_config['events'] = array_merge( (array) $tec, $this->localize() );
+
+		$post_objects                  = empty( $editor_config['post_objects'] ) ? [] : $editor_config['post_objects'];
+		$editor_config['post_objects'] = array_merge(
+			(array) $post_objects,
+			[
+				TEC::POSTTYPE => ( new Event_Object() )->data()
```
Comment (tr1b0t) [inline src/Tribe/Editor/Configuration.php:38]: * :no_entry_sign: Each line in an array declaration must end in a comma
* :no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: 391f1ffd1ff5379bb21640920288ec22</sub>

File: src/Tribe/Editor/Objects/Editor_Object_Interface.php [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+ * Class Editor_Object_Interface
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Editory\Objects
+ */
+interface Editor_Object_Interface {
+
+	/**
+	 * Returns the editor object data in the format required by the block editor.
```
Comment (tr1b0t) [inline src/Tribe/Editor/Objects/Editor_Object_Interface.php:22]: * :no_entry_sign: Doc comment for parameter "$key" missing
* :no_entry_sign: Doc comment for parameter "$default" missing

<sub>hash: d78f262eda5a06f55ffdfdb652bbf6cf</sub>

File: src/Tribe/Editor/Objects/Event.php [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+	/**
+	 * Event constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param int|\WP_Post|null $event The event post ID or object, or `null` to use the global `post` object.
+	 */
+	public function __construct( $event = null ) {
+		$event_candidate = $event ?: \tribe_get_request_var( 'post', false );
+		$this->post      = \get_post( $event_candidate );
```
Comment (tr1b0t) [inline src/Tribe/Editor/Objects/Event.php:49]: :no_entry_sign: Using short ternaries is not allowed

<sub>hash: bcdb167342df1bf7e5e941f9a829699c</sub>

File: src/Tribe/Editor/Objects/Event.php [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	/**
+	 * {@inheritDoc}
```
Comment (tr1b0t) [inline src/Tribe/Editor/Objects/Event.php:53]: * :no_entry_sign: Doc comment for parameter "$key" missing
* :no_entry_sign: Doc comment for parameter "$default" missing

<sub>hash: a566a69898ac60d6955835c5323eccd1</sub>

File: src/Tribe/Editor/Objects/Event.php [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+	 */
+	public function data( $key = null, $default = null ) {
+		if ( $this->data === null ) {
+			$this->data = [
```
Comment (tr1b0t) [inline src/Tribe/Editor/Objects/Event.php:57]: :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: ce7ce6e342a860bad3d20d7ff7a5603e</sub>

**Comment by paulmskim** [comment]
@lucatume I'm having an issue with one of the cases: if it's a new post, the `is_new_post` parameter should return `true`. With Gustavo's work in https://github.com/moderntribe/the-events-calendar/pull/3114/commits/0caf8d26e0fef23b7320d1396922eae559b00ec1, it was working as intended, but with the refactor something slipped up. Could you have a look pls?

**Pull request merged**